content/groth16/en/groth16.md

```diff
@@ -299,10 +299,10 @@ The equation above is valid, but the witness does not necessarily satisfy the or
 
 Therefore, we need to prevent the prover from using $\Psi_1$ to $\Psi_{\ell}$ as part of the computation of $[C]_1$.
 
-### Introducing $\gamma$ and $\delta$
-To avoid the problem above, the trusted setup introduces new scalars $\gamma$ and $\delta$ to force $\Psi_{\ell+1}$ to $\Psi_m$ to be separate from $\Psi_1$ to $\Psi_{\ell}$. To do this, the trusted setup divides (multiplies by the modular inverse) the private terms (that constitute $[C]_1$) by $\delta$ and the public terms (that constitute $[X]_1$, the sum the verifier computes) by $\gamma$.
+### Introducing $\gamma$ and/or $\delta$
+To avoid the problem above, the trusted setup introduces a new scalar : $\gamma$ or $\delta$ to force $\Psi_{\ell+1}$ to $\Psi_m$ to be separate from $\Psi_1$ to $\Psi_{\ell}$. To do this, the trusted setup divides (multiplies by the modular inverse) the private terms (that constitute $[C]_1$) by $\delta$ and/or the public terms (that constitute $[X]_1$, the sum the verifier computes) by $\gamma$.
 
-Since the $h(\tau)t(\tau)$ term is embedded in $[C]_1$, those terms also need to be divided by $\delta$.
+Since the $h(\tau)t(\tau)$ term is embedded in $[C]_1$, those terms also need to be divided by $\delta$. Having $\delta$ and $\gamma$ having an unknown discrete logarithm relation is enough to avoid any forgeries.
```

Comment:
Can we reword this to "If either $\delta$ or $\gamma$ have an unknown discrete log, then the forgery described earlier is avoided. In practice, $\gamma$ is simply $G_2$ and $\delta$ is randomly generated during the trusted setup. In the Tornado Cash verifier, the [`gamma2` elliptic curve point](https://github.com/tornadocash/tornado-core/blob/1ef6a263ac6a0e476d063fcb269a9df65a1bd56a/contracts/Verifier.sol#L176C25-L176C113) is the $\mathbb{G}_2$ generator. Similarly, in the [ZCash trusted setup](https://github.com/ebfull/phase2/blob/master/src/lib.rs#L808) (back when ZCash used Groth16), $\gamma$ was simply $G_2$ and $\delta$ was randomly chosen.

Reply:
@RareSkills I created this pull request with the power to let you commit what I did…
Zcash is still using Groth16 : Zcash on the technical level works as 3 cryptocurrencies : the bitcoin 1, the sapling 1 ; the orchad 1 : each of them as power of emmission in the others.

Reply:
@RareSkills Some Tornado cash keys (like classic versions deployed on other chains and nova) that don’t rely on the Zcash’s trusted setup code do have both δ and  γ.